services/rest/websocket/handler.go

```diff
@@ -121,6 +121,8 @@ func (h *Handler) Ack(conn connection.Conn, s serialization.SerializeFunc, messa
 		return nil
 	case config.Synchronous:
 		return func(err error) {
+			conn.Lock()
```

Comment:
Why do we need a `lock` in this situation? just trying to understand,
For the `async`, I don't see it.

Reply:
This is because multiple workers are trying to write messages to a single causing the race condition. Hence the lock

Reply:
I have removed this mutex as the connection object is also used in Pinger and there was some deadlock which resulted in no response from Racccoon. Have added synchronisation using channel now